keymap.c

```diff
@@ -347,82 +401,73 @@ static enum CommandResult km_bind_err(const char *s, enum MenuType menu, int op,
+  /* find position to place new keymap */
+  STAILQ_FOREACH(np, &Keymaps[menu], entries)
   {
-    if ((pos >= len) || (pos >= tmp->len))
+    compare = km_compare_keys(map, np, &pos);
+
+    if (compare == map) /* map's keycode is bigger */
     {
-      /* map and tmp match so overwrite */
-      do
-      {
-        /* Don't warn on overwriting a 'noop' binding */
-        if ((tmp->len != len) && (tmp->op != OP_NULL))
-        {
-          /* Overwrite with the different lengths, warn */
-          /* TODO: MAX_SEQ here is wrong */
-          char old_binding[MAX_SEQ];
-          char new_binding[MAX_SEQ];
-          km_expand_key(old_binding, MAX_SEQ, map);
-          km_expand_key(new_binding, MAX_SEQ, tmp);
-          if (err)
-          {
-            /* err was passed, put the string there */
-            snprintf(
-                err->data, err->dsize,
-                _("Binding '%s' will alias '%s'  Before, try: 'bind %s %s "
-                  "noop'  "
-                  "https://neomutt.org/guide/configuration.html#bind-warnings"),
-                old_binding, new_binding, mutt_map_get_name(menu, Menus), new_binding);
-          }
-          else
-          {
-            mutt_error(
-                _("Binding '%s' will alias '%s'  Before, try: 'bind %s %s "
-                  "noop'  "
-                  "https://neomutt.org/guide/configuration.html#bind-warnings"),
-                old_binding, new_binding, mutt_map_get_name(menu, Menus), new_binding);
-          }
-          rc = MUTT_CMD_WARNING;
-        }
-        len = tmp->eq;
-        next = tmp->next;
-        FREE(&tmp->macro);
-        FREE(&tmp->keys);
-        FREE(&tmp->desc);
-        FREE(&tmp);
-        tmp = next;
-      } while (tmp && len >= pos);
-      map->eq = len;
-      break;
+      last = np;
+      lastpos = pos;
+      if (pos > np->eq)
+        pos = np->eq;
     }
-    else if (buf[pos] == tmp->keys[pos])
-      pos++;
-    else if (buf[pos] < tmp->keys[pos])
+    else if (compare == np) /* np's keycode is bigger, found insert location */
     {
-      /* found location to insert between last and tmp */
       map->eq = pos;
       break;
     }
-    else /* buf[pos] > tmp->keys[pos] */
+    else /* equal keycodes */
     {
-      last = tmp;
-      lastpos = pos;
-      if (pos > tmp->eq)
-        pos = tmp->eq;
-      tmp = tmp->next;
+      /* Don't warn on overwriting a 'noop' binding */
+      if ((np->len != len) && (np->op != OP_NULL))
+      {
+        /* Overwrite with the different lengths, warn */
+        /* TODO: MAX_SEQ here is wrong */
+        char old_binding[MAX_SEQ];
+        char new_binding[MAX_SEQ];
+        km_expand_key(old_binding, MAX_SEQ, map);
+        km_expand_key(new_binding, MAX_SEQ, np);
+        if (err)
+        {
+          /* err was passed, put the string there */
+          snprintf(
+              err->data, err->dsize,
+              _("Binding '%s' will alias '%s'  Before, try: 'bind %s %s "
```

Comment:
There are two calls to gettext with identical parameters. Can you please save the result in a variable and use it in both places?

Reply:
Done. Stored result in a string called 'err_msg'.